-1

**Failed Tests**:   Build  ClangBuild 
**Size**: This PR adds an extra 36KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4f8fdc/43272/summary.html
**COMMIT**: 2cd534dde8b8b37f139a840813ded00ba82789b1
**CMSSW**: CMSSW_15_0_X_2024-12-05-0900/el8_amd64_gcc12
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46876/43272/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_15_0_X_2024-12-05-0900' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2024-12-05-0900' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/cmssw/CMSSW_15_0_X_2024-12-05-0900/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-b14784b909be2327163aa9885655770a/include/LCG -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/dd4hep/v01-29-00-f4be6b8681f14603c12e5ed4cb86972d/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/alpaka/1.1.0-aba90e6b0efd37975ff68417e953fa78/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/boost/1.80.0-933d355ba5f1bc30f82ea288aaa7b265/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/clhep/2.4.7.1-17c7283747b6cbc68a3534cb987a890c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/curl/7.79.0-abae79fca02c8e83ab0f428f2a91d96b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/gsl/2.6-4e597de26ced42091883404fa44cef2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/lcg/root/6.30.09-b9bac25074089513bc07ab14ebb67283/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tbb/v2021.9.0-e785b749a0b6cb9c66dc1d78066210e0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/OpenBLAS/0.3.27-6574458f8642b299ac965c654e194a8b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelPhase2DigiToCluster.cc.d src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc -o tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelPhase2DigiToCluster.cc.o
>> Compiling alpaka/cuda edm plugin src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_15_0_X_2024-12-05-0900' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2024-12-05-0900' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/cmssw/CMSSW_15_0_X_2024-12-05-0900/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-b14784b909be2327163aa9885655770a/include/LCG -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/dd4hep/v01-29-00-f4be6b8681f14603c12e5ed4cb86972d/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/alpaka/1.1.0-aba90e6b0efd37975ff68417e953fa78/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/boost/1.80.0-933d355ba5f1bc30f82ea288aaa7b265/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/clhep/2.4.7.1-17c7283747b6cbc68a3534cb987a890c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/curl/7.79.0-abae79fca02c8e83ab0f428f2a91d96b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/gsl/2.6-4e597de26ced42091883404fa44cef2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/lcg/root/6.30.09-b9bac25074089513bc07ab14ebb67283/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tbb/v2021.9.0-e785b749a0b6cb9c66dc1d78066210e0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/OpenBLAS/0.3.27-6574458f8642b299ac965c654e194a8b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelRawToCluster.cc.d src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc -o tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelRawToCluster.cc.o
>> Compiling alpaka/cuda src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/bin/nvcc -x cu -MMD -MF tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelRawToClusterKernel.dev.cc.d -dc -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_15_0_X_2024-12-05-0900' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2024-12-05-0900' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/cmssw/CMSSW_15_0_X_2024-12-05-0900/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-b14784b909be2327163aa9885655770a/include/LCG -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/dd4hep/v01-29-00-f4be6b8681f14603c12e5ed4cb86972d/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/alpaka/1.1.0-aba90e6b0efd37975ff68417e953fa78/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/boost/1.80.0-933d355ba5f1bc30f82ea288aaa7b265/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/clhep/2.4.7.1-17c7283747b6cbc68a3534cb987a890c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/curl/7.79.0-abae79fca02c8e83ab0f428f2a91d96b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/gsl/2.6-4e597de26ced42091883404fa44cef2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/lcg/root/6.30.09-b9bac25074089513bc07ab14ebb67283/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tbb/v2021.9.0-e785b749a0b6cb9c66dc1d78066210e0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/OpenBLAS/0.3.27-6574458f8642b299ac965c654e194a8b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02866/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include --diag-suppress 20014 -std=c++20 -O3 --generate-line-info --source-in-ptx --display-error-number --expt-relaxed-constexpr --extended-lambda -gencode arch=compute_60,code=[sm_60,compute_60] -gencode arch=compute_70,code=[sm_70,compute_70] -gencode arch=compute_75,code=[sm_75,compute_75] -gencode arch=compute_80,code=[sm_80,compute_80] -gencode arch=compute_89,code=[sm_89,compute_89] -Wno-deprecated-gpu-targets -diag-suppress=3012 -diag-suppress=3189 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -Xcudafe --gnu_version=120300 --cudart shared -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -UALPAKA_HOST_ONLY --compiler-options '-O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -std=c++20 -fPIC ' src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc -o tmp/el8_amd64_gcc12/src/RecoLocalTracker/SiPixelClusterizer/plugins/RecoLocalTrackerSiPixelClusterizerPluginsPortableCudaAsync/alpaka/SiPixelRawToClusterKernel.dev.cc.o
src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc:11:10: fatal error: DataFormats/Common/interface/AllocateForOverwrite.h: No such file or directory
   11 | #include "DataFormats/Common/interface/AllocateForOverwrite.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
src/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc:11:10: fatal error: DataFormats/Common/interface/AllocateForOverwrite.h: No such file or directory
   11 | #include "DataFormats/Common/interface/AllocateForOverwrite.h"
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package DataFormats/VertexSoA
>> Entering Package EventFilter/HcalRawToDigi
>> Entering Package HeterogeneousCore/AlpakaTest
>> Entering Package RecoLocalTracker/SiPixelClusterizer
>> Compile sequence completed for CMSSW CMSSW_15_0_X_2024-12-05-0900
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
Command exited with non-zero status 1
	Command being timed: "scram build -k -j 32 COMPILER=llvm compile BUILD_LOG=yes"
	User time (seconds): 557.13
	System time (seconds): 77.41
	Percent of CPU this job got: 1129%
</pre>

--- a/DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.cc
+++ b/DataFormats/VertexSoA/test/alpaka/ZVertexSoA_test.cc
@@ -56,11 +56,10 @@ int main() {
 
       // If the device is actually the host, use the collection as-is.
       // Otherwise, copy the data from the device to the host.
-      ZVertexHost zvertex_h;
 #ifdef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
-      zvertex_h = std::move(zvertex_d);
+      ZVertexHost zvertex_h = std::move(zvertex_d);
 #else
-      zvertex_h = cms::alpakatools::CopyToHost<ZVertexSoACollection>::copyAsync(queue, zvertex_d);
+      ZVertexHost zvertex_h = cms::alpakatools::CopyToHost<ZVertexSoACollection>::copyAsync(queue, zvertex_d);
 #endif
       alpaka::wait(queue);
       std::cout << zvertex_h.view().metadata().size() << std::endl;

--- a/EventFilter/HcalRawToDigi/plugins/alpaka/HcalDigisSoAProducer.cc
+++ b/EventFilter/HcalRawToDigi/plugins/alpaka/HcalDigisSoAProducer.cc
@@ -116,8 +116,8 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     event.emplace(digisF5HBToken_, std::move(df5_));
 
     if (qie11Digis.empty()) {
-      event.emplace(digisF01HEToken_);
-      event.emplace(digisF3HBToken_);
+      event.emplace(digisF01HEToken_, 0, event.queue());
+      event.emplace(digisF3HBToken_, 0, event.queue());
 
     } else {
       auto size_f1 = 0;

--- a/HeterogeneousCore/AlpakaTest/plugins/alpaka/TestHelperClass.cc
+++ b/HeterogeneousCore/AlpakaTest/plugins/alpaka/TestHelperClass.cc
@@ -23,8 +23,8 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     hostProductMulti2_ = portabletest::TestHostMultiCollection2{deviceProductMulti2.sizes(), iEvent.queue()};
     hostProductMulti3_ = portabletest::TestHostMultiCollection3{deviceProductMulti3.sizes(), iEvent.queue()};
 
-    alpaka::memcpy(iEvent.queue(), hostProduct_.buffer(), deviceProduct.const_buffer());
-    alpaka::memcpy(iEvent.queue(), hostProductMulti2_.buffer(), deviceProductMulti2.const_buffer());
-    alpaka::memcpy(iEvent.queue(), hostProductMulti3_.buffer(), deviceProductMulti3.const_buffer());
+    alpaka::memcpy(iEvent.queue(), hostProduct_->buffer(), deviceProduct.const_buffer());
+    alpaka::memcpy(iEvent.queue(), hostProductMulti2_->buffer(), deviceProductMulti2.const_buffer());
+    alpaka::memcpy(iEvent.queue(), hostProductMulti3_->buffer(), deviceProductMulti3.const_buffer());
   }
 }  // namespace ALPAKA_ACCELERATOR_NAMESPACE

--- a/HeterogeneousCore/AlpakaTest/plugins/alpaka/TestHelperClass.h
+++ b/HeterogeneousCore/AlpakaTest/plugins/alpaka/TestHelperClass.h
@@ -1,6 +1,8 @@
 #ifndef HeterogeneousCore_AlpakaTest_plugins_alpaka_TestHelperClass_h
 #define HeterogeneousCore_AlpakaTest_plugins_alpaka_TestHelperClass_h
 
+#include <optional>
+
 #include "DataFormats/PortableTestObjects/interface/TestHostCollection.h"
 #include "DataFormats/PortableTestObjects/interface/alpaka/TestDeviceCollection.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
@@ -23,9 +25,23 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
 
     void makeAsync(device::Event const& iEvent, device::EventSetup const& iSetup);
 
-    portabletest::TestHostCollection moveFrom() { return std::move(hostProduct_); }
-    portabletest::TestHostMultiCollection2 moveFromMulti2() { return std::move(hostProductMulti2_); }
-    portabletest::TestHostMultiCollection3 moveFromMulti3() { return std::move(hostProductMulti3_); }
+    portabletest::TestHostCollection moveFrom() {
+      auto product = std::move(*hostProduct_);
+      hostProduct_.reset();
+      return product;
+    }
+
+    portabletest::TestHostMultiCollection2 moveFromMulti2() {
+      auto product = std::move(*hostProductMulti2_);
+      hostProductMulti2_.reset();
+      return product;
+    }
+
+    portabletest::TestHostMultiCollection3 moveFromMulti3() {
+      auto product = std::move(*hostProductMulti3_);
+      hostProductMulti3_.reset();
+      return product;
+    }
 
   private:
     const device::EDGetToken<portabletest::TestDeviceCollection> getToken_;
@@ -35,9 +51,9 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     const device::ESGetToken<AlpakaESTestDataCDevice, AlpakaESTestRecordC> esTokenDevice_;
 
     // hold the output product between acquire() and produce()
-    portabletest::TestHostCollection hostProduct_;
-    portabletest::TestHostMultiCollection2 hostProductMulti2_;
-    portabletest::TestHostMultiCollection3 hostProductMulti3_;
+    std::optional<portabletest::TestHostCollection> hostProduct_;
+    std::optional<portabletest::TestHostMultiCollection2> hostProductMulti2_;
+    std::optional<portabletest::TestHostMultiCollection3> hostProductMulti3_;
   };
 }  // namespace ALPAKA_ACCELERATOR_NAMESPACE
 

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc
@@ -44,15 +44,12 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
 
     const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
     const edm::EDGetTokenT<edm::DetSetVector<PixelDigi>> pixelDigiToken_;
-
-    device::EDPutToken<SiPixelDigisSoACollection> digiPutToken_;
-    device::EDPutToken<SiPixelClustersSoACollection> clusterPutToken_;
-
-    Algo Algo_;
-
+    const device::EDPutToken<SiPixelDigisSoACollection> digiPutToken_;
+    const device::EDPutToken<SiPixelClustersSoACollection> clusterPutToken_;
     const SiPixelClusterThresholds clusterThresholds_;
-    uint32_t nDigis_ = 0;
 
+    Algo algo_;
+    uint32_t nDigis_ = 0;
     std::optional<SiPixelDigisSoACollection> digis_d_;
   };
 
@@ -88,18 +85,18 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
 
     const TrackerGeometry* geom_ = &iSetup.getData(geomToken_);
 
-    uint32_t nDigis = 0;
+    nDigis_ = 0;
     for (const auto& det : input) {
-      nDigis += det.size();
+      nDigis_ += det.size();
     }
+    digis_d_ = SiPixelDigisSoACollection(nDigis_, iEvent.queue());
 
-    if (nDigis == 0)
+    if (nDigis_ == 0)
       return;
 
-    nDigis_ = nDigis;
     SiPixelDigisHost digis_h(nDigis_, iEvent.queue());
 
-    nDigis = 0;
+    uint32_t nDigis = 0;
     for (const auto& det : input) {
       unsigned int detid = det.detId();
       DetId detIdObject(detid);
@@ -116,22 +113,20 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         ++nDigis;
       }
     }
+    assert(nDigis == nDigis_);
 
-    digis_d_ = SiPixelDigisSoACollection(nDigis, iEvent.queue());
     alpaka::memcpy(iEvent.queue(), digis_d_->buffer(), digis_h.buffer());
-    Algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d_->view(), nDigis_);
+    algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d_->view(), nDigis_);
   }
 
   void SiPixelPhase2DigiToCluster::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
     if (nDigis_ == 0) {
-      SiPixelClustersSoACollection clusters_d{pixelTopology::Phase2::numberOfModules, iEvent.queue()};
-      SiPixelDigisSoACollection digis_d_zero{0, iEvent.queue()};
-      iEvent.emplace(digiPutToken_, std::move(digis_d_zero));
-      iEvent.emplace(clusterPutToken_, std::move(clusters_d));
-    } else {
-      digis_d_->setNModules(Algo_.nModules());
       iEvent.emplace(digiPutToken_, std::move(*digis_d_));
-      iEvent.emplace(clusterPutToken_, Algo_.getClusters());
+      iEvent.emplace(clusterPutToken_, pixelTopology::Phase2::numberOfModules, iEvent.queue());
+    } else {
+      digis_d_->setNModules(algo_.nModules());
+      iEvent.emplace(digiPutToken_, std::move(*digis_d_));
+      iEvent.emplace(clusterPutToken_, algo_.getClusters());
     }
     digis_d_.reset();
   }

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelPhase2DigiToCluster.cc
@@ -1,5 +1,6 @@
 // C++ includes
 #include <memory>
+#include <optional>
 #include <stdexcept>
 #include <string>
 #include <utility>
@@ -52,7 +53,7 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     const SiPixelClusterThresholds clusterThresholds_;
     uint32_t nDigis_ = 0;
 
-    SiPixelDigisSoACollection digis_d;
+    std::optional<SiPixelDigisSoACollection> digis_d_;
   };
 
   SiPixelPhase2DigiToCluster::SiPixelPhase2DigiToCluster(const edm::ParameterSet& iConfig)
@@ -88,7 +89,6 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     const TrackerGeometry* geom_ = &iSetup.getData(geomToken_);
 
     uint32_t nDigis = 0;
-
     for (const auto& det : input) {
       nDigis += det.size();
     }
@@ -107,39 +107,33 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
       auto const gind = genericDet->index();
       for (auto const& px : det) {
         digis_h.view()[nDigis].moduleId() = uint16_t(gind);
-
         digis_h.view()[nDigis].xx() = uint16_t(px.row());
         digis_h.view()[nDigis].yy() = uint16_t(px.column());
         digis_h.view()[nDigis].adc() = uint16_t(px.adc());
-
         digis_h.view()[nDigis].clus() = 0;
-
         digis_h.view()[nDigis].pdigi() = uint32_t(px.packedData());
-
         digis_h.view()[nDigis].rawIdArr() = uint32_t(detid);
-
-        nDigis++;
+        ++nDigis;
       }
     }
 
-    digis_d = SiPixelDigisSoACollection(nDigis, iEvent.queue());
-    alpaka::memcpy(iEvent.queue(), digis_d.buffer(), digis_h.buffer());
-
-    Algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d.view(), nDigis_);
+    digis_d_ = SiPixelDigisSoACollection(nDigis, iEvent.queue());
+    alpaka::memcpy(iEvent.queue(), digis_d_->buffer(), digis_h.buffer());
+    Algo_.makePhase2ClustersAsync(iEvent.queue(), clusterThresholds_, digis_d_->view(), nDigis_);
   }
 
   void SiPixelPhase2DigiToCluster::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
     if (nDigis_ == 0) {
       SiPixelClustersSoACollection clusters_d{pixelTopology::Phase2::numberOfModules, iEvent.queue()};
-      SiPixelDigisSoACollection digis_d_zero{nDigis_, iEvent.queue()};
+      SiPixelDigisSoACollection digis_d_zero{0, iEvent.queue()};
       iEvent.emplace(digiPutToken_, std::move(digis_d_zero));
       iEvent.emplace(clusterPutToken_, std::move(clusters_d));
-      return;
+    } else {
+      digis_d_->setNModules(Algo_.nModules());
+      iEvent.emplace(digiPutToken_, std::move(*digis_d_));
+      iEvent.emplace(clusterPutToken_, Algo_.getClusters());
     }
-
-    digis_d.setNModules(Algo_.nModules());
-    iEvent.emplace(digiPutToken_, std::move(digis_d));
-    iEvent.emplace(clusterPutToken_, Algo_.getClusters());
+    digis_d_.reset();
   }
 
 }  // namespace ALPAKA_ACCELERATOR_NAMESPACE

--- a/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc
+++ b/RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc
@@ -1,4 +1,5 @@
 #include <memory>
+#include <optional>
 #include <stdexcept>
 #include <string>
 #include <utility>
@@ -262,10 +263,10 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
       // are no valid pointers to clusters' Collection columns, instantiation
       // of TrackingRecHits fail. Example: workflow 11604.0
 
-      iEvent.emplace(digiPutToken_, nDigis_, iEvent.queue());
+      iEvent.emplace(digiPutToken_, 0, iEvent.queue());
       iEvent.emplace(clusterPutToken_, pixelTopology::Phase1::numberOfModules, iEvent.queue());
       if (includeErrors_) {
-        iEvent.emplace(digiErrorPutToken_);
+        iEvent.emplace(digiErrorPutToken_, 0, iEvent.queue());
         iEvent.emplace(fmtErrorToken_);
       }
       return;

